mobile/src/main/java/org/openhab/habdroid/ui/WidgetListFragment.kt

```diff
@@ -140,41 +154,118 @@ class WidgetListFragment : Fragment(), WidgetAdapter.ItemClickListener {
+
         if (widget.linkedPage != null) {
-            labels.add(getString(R.string.nfc_action_to_sitemap_page))
+            if (nfcSupported) {
+                if (hasCommandOptions) {
+                    val nfcMenu = menu.addSubMenu(Menu.NONE, 1000, Menu.NONE, R.string.nfc_action_write_command_tag)
+                    nfcMenu.setHeaderTitle(R.string.item_picker_dialog_title)
+                    populateNfcStatesMenu(nfcMenu, context, suggestedCommands, widget)
+                }
+                menu.add(Menu.NONE, 1001, Menu.NONE, R.string.nfc_action_to_sitemap_page)
+            }
             if (ShortcutManagerCompat.isRequestPinShortcutSupported(context)) {
-                labels.add(getString(R.string.home_shortcut_pin_to_home))
+                menu.add(Menu.NONE, 1002, Menu.NONE, R.string.home_shortcut_pin_to_home)
             }
+        } else if (hasCommandOptions && nfcSupported) {
+            menu.setHeaderTitle(R.string.nfc_action_write_command_tag)
+            populateNfcStatesMenu(menu, context, suggestedCommands, widget)
         }
+    }
 
-        if (labels.isEmpty()) {
-            return false
-        }
+    private fun isEmulator(): Boolean {
+        return Build.FINGERPRINT.startsWith("generic") ||
+            Build.FINGERPRINT.startsWith("unknown") ||
+            Build.MODEL.contains("google_sdk") ||
+            Build.MODEL.contains("Emulator") ||
+            Build.MODEL.contains("Android SDK built for x86") ||
+            Build.MANUFACTURER.contains("Genymotion") ||
+            (Build.BRAND.startsWith("generic") && Build.DEVICE.startsWith("generic")) ||
+            "google_sdk" == Build.PRODUCT
+    }
 
-        AlertDialog.Builder(context)
-            .setTitle(R.string.nfc_dialog_title)
-            .setItems(labels.toTypedArray()) { _, which ->
-                if (which == commands.size + 1 && widget.linkedPage != null) {
-                    createShortcut(context, widget.linkedPage)
-                } else {
-                    val itemToHandle = if (which < commands.size) widget.item else null
-                    val linkToHandle = if (which == commands.size) widget.linkedPage?.link else null
-                    if (itemToHandle != null) {
-                        startActivity(WriteTagActivity.createItemUpdateIntent(context,
-                            itemToHandle.name, commands[which], labels[which], itemToHandle.label.orEmpty()))
-                    } else if (linkToHandle != null) {
-                        startActivity(WriteTagActivity.createSitemapNavigationIntent(context, linkToHandle))
+    private fun populateNfcStatesMenu(
+        menu: Menu,
+        context: Context,
+        suggestedCommands: SuggestedCommandsFactory.SuggestedCommands,
+        widget: Widget
+    ) {
+        val name = widget.item?.name ?: return
+        val listener = object: MenuItem.OnMenuItemClickListener {
```

Comment:
False positive?

Reply:
You mean the Stickler error? Doesn't look like a false positive to me, as there's a rule to put a space before and after the colon in the `DerivedClassName : BaseClassName` syntax.